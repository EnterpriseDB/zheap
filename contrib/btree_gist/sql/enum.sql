-- enum check

create type rainbow as enum ('r','o','y','g','b','i','v');

CREATE TABLE enumtmp (a rainbow);

\copy enumtmp from 'data/enum.data'

SET enable_seqscan=on;

select a, count(*) from enumtmp group by a order by 1;

SELECT count(*) FROM enumtmp WHERE a <  'g'::rainbow;

SELECT count(*) FROM enumtmp WHERE a <= 'g'::rainbow;

SELECT count(*) FROM enumtmp WHERE a  = 'g'::rainbow;

SELECT count(*) FROM enumtmp WHERE a >= 'g'::rainbow;

SELECT count(*) FROM enumtmp WHERE a >  'g'::rainbow;

CREATE INDEX enumidx ON enumtmp USING gist ( a );

SET enable_seqscan=off;
SET enable_bitmapscan=off;
SET enable_indexonlyscan=off;

SELECT count(*) FROM enumtmp WHERE a <  'g'::rainbow;

SELECT count(*) FROM enumtmp WHERE a <= 'g'::rainbow;

SELECT count(*) FROM enumtmp WHERE a  = 'g'::rainbow;

SELECT count(*) FROM enumtmp WHERE a >= 'g'::rainbow;

SELECT count(*) FROM enumtmp WHERE a >  'g'::rainbow;

EXPLAIN (COSTS OFF)
SELECT count(*) FROM enumtmp WHERE a >= 'g'::rainbow;
