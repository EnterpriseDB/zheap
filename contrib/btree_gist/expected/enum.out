-- enum check
create type rainbow as enum ('r','o','y','g','b','i','v');
CREATE TABLE enumtmp (a rainbow);
\copy enumtmp from 'data/enum.data'
SET enable_seqscan=on;
select a, count(*) from enumtmp group by a order by 1;
 a | count 
---+-------
 r |    76
 o |    78
 y |    73
 g |    75
 b |    77
 i |    78
 v |    75
   |    63
(8 rows)

SELECT count(*) FROM enumtmp WHERE a <  'g'::rainbow;
 count 
-------
   227
(1 row)

SELECT count(*) FROM enumtmp WHERE a <= 'g'::rainbow;
 count 
-------
   302
(1 row)

SELECT count(*) FROM enumtmp WHERE a  = 'g'::rainbow;
 count 
-------
    75
(1 row)

SELECT count(*) FROM enumtmp WHERE a >= 'g'::rainbow;
 count 
-------
   305
(1 row)

SELECT count(*) FROM enumtmp WHERE a >  'g'::rainbow;
 count 
-------
   230
(1 row)

CREATE INDEX enumidx ON enumtmp USING gist ( a );
SET enable_seqscan=off;
SET enable_bitmapscan=off;
SET enable_indexonlyscan=off;
SELECT count(*) FROM enumtmp WHERE a <  'g'::rainbow;
 count 
-------
   227
(1 row)

SELECT count(*) FROM enumtmp WHERE a <= 'g'::rainbow;
 count 
-------
   302
(1 row)

SELECT count(*) FROM enumtmp WHERE a  = 'g'::rainbow;
 count 
-------
    75
(1 row)

SELECT count(*) FROM enumtmp WHERE a >= 'g'::rainbow;
 count 
-------
   305
(1 row)

SELECT count(*) FROM enumtmp WHERE a >  'g'::rainbow;
 count 
-------
   230
(1 row)

EXPLAIN (COSTS OFF)
SELECT count(*) FROM enumtmp WHERE a >= 'g'::rainbow;
                QUERY PLAN                 
-------------------------------------------
 Aggregate
   ->  Index Scan using enumidx on enumtmp
         Index Cond: (a >= 'g'::rainbow)
(3 rows)

